File: src/pocketcloud/network/packet/impl/normal/LibrarySyncPacket.php
<?php

namespace pocketcloud\network\packet\impl\normal;

use pocketcloud\library\Library;
use pocketcloud\library\LibraryManager;
use pocketcloud\network\client\ServerClient;
use pocketcloud\network\packet\CloudPacket;
use pocketcloud\network\packet\utils\PacketData;

class LibrarySyncPacket extends CloudPacket {

    private array $data = [];

    public function __construct() {
        foreach (array_filter(LibraryManager::getInstance()->getLibraries(), fn(Library $library) => $library->isCloudBridgeOnly()) as $lib) {
            $this->data[] = [
                "name" => $lib->getName(),
                "path" => $lib->getUnzipLocation()
            ];
        }
    }

    public function encodePayload(PacketData $packetData) {
        $packetData->write($this->data);
    }

    public function decodePayload(PacketData $packetData) {
        $this->data = $packetData->readArray();
    }

    public function getData(): array {
        return $this->data;
    }

    public function handle(ServerClient $client) {}
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/LibrarySyncPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.network.packet.CloudPacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Collection;
import java.util.LinkedHashMap;
import java.util.List;

@NoArgsConstructor
@Getter
public final class LibrarySyncPacket extends CloudPacket implements ClientboundPacket {

    private Collection<LinkedHashMap<String, String>> data;

    public LibrarySyncPacket(Collection<LinkedHashMap<String, String>> data) {
        this.data = data != null ? data : List.of();
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(data);
    }

    @Override
    public void decodePayload(IPacketData packetData) {}

    public static LibrarySyncPacket create(Collection<LinkedHashMap<String, String>> data) {
        return new LibrarySyncPacket(data);
    }
}